### ISSUE TITLE
Library rename fails with libraries using namespaces.

### ISSUE THREAD

**Issue by arturomichel** [manahl/arctic#343]
#### Arctic Version
```
1.37
```

#### Arctic Store
```
VersionStore, TickStore
```

#### Platform and version
Ubuntu 14.04.5 LTS
Python 3.5
Mongo 3.2.9

#### Description of problem and/or code sample that reproduces the issue
The rename feature throws an error when using namespaces. There are a couple of other scenarios under which this fails. I can only get it to work with libraries with no namespace. 

```
import arctic as a
store = a.Arctic('localhost')
store.initialize_library('test.TEST_LIBRARY')
store.rename_library('test.TEST_LIBRARY', 'prod.TEST_LIBRARY')
---------------------------------------------------------------------------
OperationFailure                          Traceback (most recent call last)
<ipython-input-14-ca192b90bd53> in <module>()
----> 1 lib =get._store.rename_library('test.TEST_LIBRARY', 'prod.TEST_LIBRARY')

/opt/anaconda3/envs/dev1/lib/python3.5/site-packages/arctic/arctic.py in rename_library(self, from_lib, to_lib)
    320         for coll in l._db.collection_names():
    321             if coll.startswith(colname + '.'):
--> 322                 l._db[coll].rename(coll.replace(from_lib, to_lib))
    323         if from_lib in self._library_cache:
    324             del self._library_cache[from_lib]

/opt/anaconda3/envs/dev1/lib/python3.5/site-packages/pymongo/collection.py in rename(self, new_name, **kwargs)
   1773         cmd.update(kwargs)
   1774         with self._socket_for_writes() as sock_info:
-> 1775             sock_info.command('admin', cmd)
   1776 
   1777     def distinct(self, key, filter=None, **kwargs):

/opt/anaconda3/envs/dev1/lib/python3.5/site-packages/pymongo/pool.py in command(self, dbname, spec, slave_ok, read_preference, codec_options, check, allowable_errors, check_keys, read_concern)
    237                            check, allowable_errors, self.address,
    238                            check_keys, self.listeners, self.max_bson_size,
--> 239                            read_concern)
    240         except OperationFailure:
    241             raise

/opt/anaconda3/envs/dev1/lib/python3.5/site-packages/pymongo/network.py in command(sock, dbname, spec, slave_ok, is_mongos, read_preference, codec_options, check, allowable_errors, address, check_keys, listeners, max_bson_size, read_concern)
    100         response_doc = unpacked['data'][0]
    101         if check:
--> 102             helpers._check_command_response(response_doc, None, allowable_errors)
    103     except Exception as exc:
    104         if publish:

/opt/anaconda3/envs/dev1/lib/python3.5/site-packages/pymongo/helpers.py in _check_command_response(response, msg, allowable_errors)
    203 
    204             msg = msg or "%s"
--> 205             raise OperationFailure(msg % errmsg, code, response)
    206 
    207 

OperationFailure: target namespace exists
```

**Comment by bmoscon** [comment]
it should work if you dont change the namespace name, and change only the library name - i.e. if you went from 'test.TEST_LIBRARY' to 'test.NEW_LIB_NAME'

I can put a change in to detect this earlier and fail, but it will still fail. Is namespace renaming something you want/need?

**Comment by arturomichel** [comment]
We are/were thinking to use the rename feature to "upgrade" our data sets.

For instance, if we wanted to add or modify something in a dataset, we want to isolate it from the typical users while we test it and backfill any missing data. Once the acceptance process is complete, we thought we could move it to a different namespace which is available to the final users.

Have you come across this use case before? Perhaps there is a better way of addressing it.

**Comment by bmoscon** [comment]
Ah ok. Renaming isnt any better than copying from one library to another (in terms of speed, its probably easier). I'd just read every symbol and copy it to a new library. something like:

```python
for symbol in library.get_symbols():
    data = lib.read(symbol)
    new_lib.write(symbol, data)
```

That being said, I will look into doing a complete rename. It should be possible, its just going to take a code change. I probably wont get to this until the weekend.

**Comment by arturomichel** [comment]
We'll take this approach. It's not a very common or time sensitive operation so speed is not a major concern. 
Thanks.

**Comment by bmoscon** [comment]
@arturomichel thanks for updating. I do plan on fixing this, but since I know its not a priority, it'll be a little while :)

**Comment by arturomichel** [comment]
One gotcha with this is to read the complete dataset for each symbol. By default, ```data = lib.read(symbol)``` retrieves only a portion of the data and you'll end up with an incomplete copy. ``` data = lib.read(symbol, date_rage=DateRange(beginning_of_time, end_of_time)``` worked for me.

**Comment by bmoscon** [comment]
oh ok, so you're reading from tickstore and writing to version store. That makes sense

**Comment by bmoscon** [comment]
resolved
